CM.Text/BusinessMessaging/Model/MultiChannel/WhatsAppTemplate.cs

```diff
@@ -168,19 +153,22 @@ public class TemplateDateTimeComponent
         public int Minute { get; }
 
         /// <summary>
-        /// Constructor initializing the values.
+        /// Constructor initializing the component.
         /// </summary>
         /// <param name="moment"></param>
-        public TemplateDateTimeComponent(DateTime moment)
+        public TemplateDateTime(DateTime moment)
         {
+            FallbackValue = moment.ToUniversalTime().ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fff'Z'");
             Minute = moment.Minute;
             Hour = moment.Hour;
             Month = moment.Month;
             Year = moment.Year;
             DayOfMonth = moment.Day;
-            DayOfWeek = moment.DayOfWeek.ToString();
+            DayOfWeek = (int)moment.DayOfWeek;
```

Comment:
Did you check WhatsApp also handles the dayofweek starting with zero == sunday or monday etc?

Reply:
I am glad you asked because via Facebook/WhatsApp specs 

day_of_week: Options: "MONDAY", 1, "TUESDAY", 2, "WEDNESDAY", 3, "THURSDAY", 4, "FRIDAY", 5, "SATURDAY", 6, "SUNDAY", 7

Code has been updated to use the numbers Facebook expect